packages/interactive_media_ads/lib/src/platform_interface/interactive_media_ads_platform.dart

```diff
@@ -0,0 +1,63 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:plugin_platform_interface/plugin_platform_interface.dart';
+
+import 'platform_ad_display_container.dart';
+import 'platform_ads_loader.dart';
+import 'platform_ads_manager_delegate.dart';
+
+/// Interface for a platform implementation of the Interactive Media Ads SDKs.
+abstract class InteractiveMediaAdsPlatform extends PlatformInterface {
+  /// Creates a new [InteractiveMediaAdsPlatform].
+  InteractiveMediaAdsPlatform() : super(token: _token);
+
+  static final Object _token = Object();
+
+  static InteractiveMediaAdsPlatform? _instance;
+
+  /// The instance of [InteractiveMediaAdsPlatform] to use.
+  static InteractiveMediaAdsPlatform? get instance => _instance;
+
+  /// Platform-specific plugins should set this with their own platform-specific
+  /// class that extends [InteractiveMediaAdsPlatform] when they register
+  /// themselves.
+  static set instance(InteractiveMediaAdsPlatform? instance) {
+    if (instance == null) {
+      throw AssertionError(
+        'Platform interfaces can only be set to a non-null instance',
```

Comment:
Why? We had an issue filed about this in another plugin; if the instance can be nullable anyway, why not allow nulling it?